ios/FluentUI/Vnext/ActivityIndicator/ActivityIndicator.swift

```diff
@@ -0,0 +1,153 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+import SwiftUI
+
+/// Properties available to customize the state of the activity indicator state
+@objc public protocol MSFActivityIndicatorState {
+    var accessibilityLabel: String? { get set }
+    var color: UIColor? { get set }
+    var isAnimating: Bool { get set }
+    var hidesWhenStopped: Bool { get set }
+    var size: MSFActivityIndicatorSize { get set }
+}
+
+/// Properties available to customize the state of the activity indicator
+class MSFActivityIndicatorStateImpl: NSObject, ObservableObject, MSFActivityIndicatorState {
+    @Published var color: UIColor?
+    @Published var isAnimating: Bool = false
+    @Published var hidesWhenStopped: Bool = true
+
+    let tokens: MSFActivityIndicatorTokens
+
+    var size: MSFActivityIndicatorSize {
+        get {
+            return tokens.size
+        }
+
+        set {
+            tokens.size = newValue
+        }
+    }
+
+    init(size: MSFActivityIndicatorSize) {
+        self.tokens = MSFActivityIndicatorTokens(size: size)
+    }
+}
+
+/// View that represents the activity indicator
+public struct ActivityIndicator: View {
+    @Environment(\.theme) var theme: FluentUIStyle
+    @Environment(\.windowProvider) var windowProvider: FluentUIWindowProvider?
+    @ObservedObject var tokens: MSFActivityIndicatorTokens
+    @ObservedObject var state: MSFActivityIndicatorStateImpl
+    @State var rotationAngle: Double = 0.0
+
+    public init(size: MSFActivityIndicatorSize) {
+        let state = MSFActivityIndicatorStateImpl(size: size)
+        self.state = state
+        self.tokens = state.tokens
+    }
+
+    public var body: some View {
+        if state.isAnimating {
+            animatedSemiRing
+        } else if state.hidesWhenStopped {
+            EmptyView()
+        } else {
+            semiRing
+        }
+    }
+
+    private var animatedSemiRing: some View {
+        return semiRing
+            .rotationEffect(Angle(degrees: rotationAngle))
+            .onAppear {
+                rotationAngle = initialAnimationRotationAngle
+
+                withAnimation(Animation.linear(duration: animationDuration)
+                                .repeatForever(autoreverses: false)) {
```

Comment:
Good to know it's a linear animation! I think other ones I've seen do some complex stuff with easing over the course of the animation 😅